### ISSUE TITLE
Documenting `customizations` and namespace contribution process

### ISSUE THREAD

**Issue by bamurtaugh** [devcontainers/spec#80]
We have some info about contributing a new namespace and properties for a tool [here](https://github.com/devcontainers/spec/blob/main/contributing.md#contributing-tool-specific-support), but it's not very detailed.

- When should a tool/org/company propose a new namespace?
- What does the contribution process look like (i.e. different detailed steps for publicly vs privately contributed namespaces)?

**Comment by Chuxel** [comment]
@avidal Yeah, adding a reverse DNS name under `customizations` is a safe way to avoid naming conflicts. Love to hear what you are looking into as well!
